flink-connectors/flink-connector-filesystem/src/main/java/org/apache/flink/streaming/connectors/fs/bucketing/BucketingSink.java

```diff
@@ -908,6 +929,20 @@ private void handlePendingFilesForPreviousCheckpoints(Map<Long, List<String>> pe
 		return this;
 	}
 
+	/**
+	 * Sets the roll over interval in milliseconds.
+	 *
+	 *
+	 * <p>When a bucket part file is older than the roll over interval, a new bucket part file is
+	 * started and the old one is closed. The name of the bucket file depends on the {@link Bucketer}.
+	 *
+	 * @param batchRolloverInterval The roll over interval in milliseconds
+	 */
+	public BucketingSink<T> setBatchRolloverInterval(long batchRolloverInterval) {
+		this.batchRolloverInterval = batchRolloverInterval;
+		return this;
```

Comment:
Please add checks for invalid configs like negative values.

Reply:
Added a check for `batchRolloverInterval` to be a positive non-zero value.